### PR TITLE
Remove RMSE function

### LINKED ISSUE #3982: pybamm.rmse. document return type and args types
### PyBaMM Version

23.9

### Python Version

3.11.7

### Describe the bug

Minor issue for the PyBaMM training workshop

### Steps to Reproduce

Read the documentation of pybamm.rmse

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by kratman**
# Description

Minor removal of code and some cleanup of IDE warnings

Fixes #3982

## Type of change

Removes old code

# Key checklist:

- [x] No style issues: `$ pre-commit run` (or `$ nox -s pre-commit`) (see [CONTRIBUTING.md](https://github.com/pybamm-team/PyBaMM/blob/develop/CONTRIBUTING.md#installing-and-using-pre-commit) for how to set this up to run automatically when committing locally, in just two lines of code)
- [x] All tests pass: `$ python run-tests.py --all` (or `$ nox -s tests`)
- [x] The documentation builds: `$ python run-tests.py --doctest` (or `$ nox -s doctests`)

You can run integration tests, unit tests, and doctests together at once, using `$ python run-tests.py --quick` (or `$ nox -s quick`).

## Further checks:

- [x] Code is commented, particularly in hard-to-understand areas
- [x] Tests added that prove fix is effective or that feature works

### CODE REVIEW

File: pybamm/__init__.py [hunk 1]
```diff
@@ -214,38 +160,38 @@
...
+FLOAT_FORMAT = "{: .17e}"
+# Absolute path to the PyBaMM repo
+script_path = os.path.abspath(__file__)
+
+ABSOLUTE_PATH = root_dir()
+PARAMETER_PATH = [
+    root_dir(),
+    os.getcwd(),
+    os.path.join(root_dir(), "pybamm", "input", "parameters"),
+]
```
Comment (kratman) [inline pybamm/__init__.py:194]: @tinosulzer These don't appear to be used anywhere are they OK to remove as well?

**Review by tinosulzer — CHANGES_REQUESTED** [review]
All the code removed is fine. Please don't remove those comments. They add structure to the code and make it easier to scan. Removing the empty comment lines (`#` and nothing else) is fine.

**Comment by kratman** [comment]
> All the code removed is fine. Please don't remove those comments. They add structure to the code and make it easier to scan. Removing the empty comment lines (`#` and nothing else) is fine. 

I removed them because they basically all said something along the lines of
```python
# functions from file X
from x import y
```
Which does not seem to add anything

**Comment by tinosulzer** [comment]
It adds structure and makes the file easier to scan since they are highlighted in different colors by most IDEs

**Comment by Saransh-cpp** [comment]
Imports in pybamm's `__init__.py` file must be in a particular order (autoformatting tools usually mess up this order), so adding a structure with the comments makes sense.

**Comment by kratman** [comment]
> It adds structure and makes the file easier to scan since they are highlighted in different colors by most IDEs

To me it did not add anything since it literally said what was right below that in the code, but I will add it back if you insist. In general my changes like this are just trying to clean up stuff that does not add anything like:
```python
#
# Batch Study
#
from .batch_study import BatchStudy
```
which ends up adding a lot of comments for the exact thing that is below it.

> Imports in pybamm's `__init__.py` file must be in a particular order (autoformatting tools usually mess up this order), so adding a structure with the comments makes sense.

I did not run any auto-formatting here and left the imports in groups. The only thing I did for my IDE was move the constants to the bottom of the file in order to stop the warnings about imports coming after code. The comments did not add any hints about which order things needed to be imported, no warnings about problematic stuff, etc. So I don't think they solve the import order problem either

However, the constants I moved look like they are not actually used and can probably be removed as well. PyBaMM triggers so many warnings in PyCharm that I often cannot see the real problems from the inspections. I try to clean up what I can as I work

**Comment by kratman** [comment]
Ok everything here is passing now besides Lychee

**Review by valentinsulzer — APPROVED** [review]

**Pull request merged**